pkg/provisioner/provisioner.go

```diff
@@ -68,9 +69,14 @@ func NewZFSProvisioner(parent *zfs.Dataset, shareOptions string, serverHostname
 		kubernetesReclaimPolicy = v1.PersistentVolumeReclaimRetain
 	}
 
+	if doNfsExport {
```

Comment:
This logic is backwards. doNfsExport will be true when we want to share via nfs and false when we don't